app/src/main/java/com/aerogear/androidshowcase/providers/PushServiceProvider.java

```diff
@@ -0,0 +1,57 @@
+package com.aerogear.androidshowcase.providers;
+
+import com.aerogear.androidshowcase.R;
+
+import org.aerogear.mobile.core.MobileCore;
+import org.aerogear.mobile.core.executor.AppExecutors;
+import org.aerogear.mobile.core.reactive.Responder;
+import org.aerogear.mobile.push.PushService;
+import org.aerogear.mobile.push.UnifiedPushConfig;
+
+import java.util.Arrays;
+
+public class PushServiceProvider {
+
+    private static PushServiceProvider instance;
+
+    private PushService pushService;
+    private Exception registrationException;
+
+    protected PushServiceProvider() {
+        pushService = new PushService.Builder().openshift().build();
+    }
+
+    /**
+     * Register the push service.
+     */
+    public void registerDevice() {
+        pushService.registerDevice().respondOn(new AppExecutors().mainThread())
```

Comment:
This probably shouldn't be called back on the main thread since it doesn't affect UI code.  There's nothing wrong with it per se, but if you could use the other executor (the one that isn't a network) that would be nicer.